contracts/standard/proxy/ArbitratorVersioningProxy.sol

```diff
@@ -128,11 +128,11 @@ contract ArbitratorVersioningProxy is Arbitrator, Arbitrable, VersioningProxy {
     }
 
     /** @dev Execute a ruling of a dispute.
-     *  @param _externalDisputeID ID of the dispute in the Arbitrator contract.
+     *  @param _localDisputeID ID of the dispute in the Arbitrator contract.
      *  @param _ruling Ruling given by the arbitrator. Note that 0 is reserved for "Not able/wanting to make a decision".
      */
-    function executeRuling(uint _externalDisputeID, uint _ruling) internal {
-        disputes[externalDisputeIDToLocalDisputeID[_externalDisputeID]].arbitrated.rule(_externalDisputeID, _ruling);
+    function executeRuling(uint _localDisputeID, uint _ruling) internal {
+        disputes[_localDisputeID].arbitrated.rule(disputes[_localDisputeID].externalDisputeID, _ruling);
```

Comment:
You are doing the same thing except now you expect the local dispute ID, which you won't get, so now it's double broken.

This function:
- Receives the external dispute ID, because it's an arbitrable contract to the external arbitrator.
- Finds the local dispute ID and uses it to rule on the arbitrated contract because it's an arbitrator contract to it.

Reply:
Like this @epiqueras ?

```
    function executeRuling(uint _externalDisputeID, uint _ruling) internal {
        uint localDisputeID = externalDisputeIDToLocalDisputeID[_externalDisputeID];
        disputes[localDisputeID].arbitrated.rule(localDisputeID, _ruling);
    }
```

Reply:
Yeah